backends/bmv2/header.cpp

```diff
@@ -223,19 +223,21 @@ void ConvertHeaders::addHeaderType(const IR::Type_StructLike *st) {
     }
     json->add_header_type(name, fields, max_length_bytes);
 
-    LOG1("... creating aliases for metadata fields " << st);
-    for (auto f : st->fields) {
-        if (auto name_annotation = f->getAnnotation("alias")) {
-            auto container = new Util::JsonArray();
-            auto alias = new Util::JsonArray();
-            auto target_name = name_annotation->expr.front()->to<IR::StringLiteral>()->value;
-            LOG2("field alias " << target_name);
-            container->append(target_name);  // name on target
-            // break down the alias into meta . field
-            alias->append(name);      // metadata name
-            alias->append(f->name);   // field name
-            container->append(alias);
-            json->field_aliases->append(container);
+    if (options.langVersion == BMV2::BMV2Options::FrontendVersion::P4_16) {
```

Comment:
There's a shortcut to this test (in fact the complement of it) called CompilerOptions::isv1().

Reply:
I don't really understand how this fixes the problem when one converts a P4-14 program to a P4-16 program and then tries to compile the latter. Won't this cause the broadcast to not work?